ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml

```diff
@@ -114,6 +114,7 @@ SQUtils.QObject {
                 if (session === null)
                     return
                 root.displayToastMessage(qsTr("Failed to authenticate %1 from %2").arg(methodStr).arg(session.peer.metadata.url), true)
+                root.sessionRequestResult(request, false /*isSuccessful*/)
```

Comment:
@alexjba this is nice, but can we do the same if the user cancels the "Sign" request? That happens before the authentication popup is run.

Reply:
We do. We reject any sign request if the sign modal is closed in any other way except when hitting `Sign`